cachetory/decorators/async_.py

```diff
@@ -32,18 +32,19 @@ def cached(
             and the rest of the arguments next to it.
         make_key: callable to generate a custom cache key per each call.
         if_not_exists: controls concurrent sets: if `True` – avoids overwriting a cached value.
-        time_to_live: cached value expiration time.
+        time_to_live: cached value expiration time or async callable that returns the expiration time.
     """
 
     def wrap(callable_: Callable[P, Awaitable[ValueT]]) -> Callable[P, Awaitable[ValueT]]:
         @wraps(callable_)
         async def cached_callable(*args: P.args, **kwargs: P.kwargs) -> ValueT:
             cache_ = await cache(callable_, *args, **kwargs) if callable(cache) else cache
+            time_to_live_ = await time_to_live() if callable(time_to_live) else time_to_live
```

Comment:
I made it accept no arguments at all. We could pass through some if needed. You @eigenein mentioned elsewhere the cache `key`. Shall I do that? Any idea of which other args can be useful? Same PR or separate?

Reply:
Let's pass the key for now, and think about other possibilities later 👍

Reply:
Done. The key is forwarded now. Typing is a bit ehh, since the `Callable[..., Awaitable[timedelta]]]]` does not really capture the information that `kwargs` are needed (right?). We could type it with a callable protocol or something, but don't know what you prefer or if it's ok as is.

Reply:
No, it doesn't. `…` means «anything». Let's merge and release to continue the work, I'll address the typing later